tgui/packages/tgui/interfaces/Sensors.tsx

```diff
@@ -0,0 +1,598 @@
+          <Table.Row>
+            <Table.Cell>{data.deep_scan_name}:</Table.Cell>
+            <Table.Cell>
+              Effective Range: {data.deep_scan_range}
+              {', '}
+              {data.deep_scan_toggled ? 'Active' : 'Inactive'}
+              {', '}
+              <Button
+                content={'on/off'}
+                onClick={() =>
+                  act('deep_scan_toggle', { deep_scan_toggle: true })
+                }
+              />
+            </Table.Cell>
+          </Table.Row>
+        ) : (
+          ''
+        )}
+        <Table.Row>
+          <Table.Cell>Integrity:</Table.Cell>
+          <Table.Cell>
+            <ProgressBar
+              animated
+              backgroundColor={Color.lerp(
+                Color.fromHex('#20b142'),
+                Color.fromHex('#db2828'),
+                data.health / data.max_health
+              )}
+              color={(() => {
+                if (data.health > (data.max_health / 3) * 2) {
+                  return 'green';
+                } else if (data.health > (data.max_health / 3) * 1) {
+                  return 'yellow';
+                } else {
+                  return 'red';
+                }
+              })()}
+              minValue={0}
+              maxValue={data.max_health}
+              value={data.health}>
+              {data.health} / {data.max_health}
+            </ProgressBar>
+          </Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Temperature:</Table.Cell>
+          <Table.Cell>
+            <ProgressBar
+              animated
+              backgroundColor={Color.lerp(
+                Color.fromHex('#20b142'),
+                Color.fromHex('#db2828'),
+                data.heat / data.critical_heat
+              )}
+              color={(() => {
+                if (data.heat > (data.critical_heat / 3) * 2) {
+                  return 'red';
+                } else if (data.heat > (data.critical_heat / 3) * 1) {
+                  return 'yellow';
+                } else {
+                  return 'green';
+                }
+              })()}
+              minValue={0}
+              maxValue={data.critical_heat}
+              value={data.heat}>
+              {data.heat} / {data.critical_heat}
+            </ProgressBar>
+          </Table.Cell>
+        </Table.Row>
+      </Table>
+    </Section>
+  );
+};
+
+const ContactsSection = function (act, data: SensorsData) {
+  return (
+    <Section title="Sensor Contacts">
+      {data.contacts && data.contacts.length ? (
+        <Table>
+          <Table.Row header>
+            <Table.Cell />
+            <Table.Cell title="Designation">Designation</Table.Cell>
+            <Table.Cell title="Bearing">B</Table.Cell>
+            <Table.Cell title="X Grid Coordinate">X</Table.Cell>
+            <Table.Cell title="Y Grid Coordinate">Y</Table.Cell>
+            <Table.Cell title="Distance">D</Table.Cell>
```

Comment:
Might be useful to tooltip this and explain what "one distance" is

Reply:
done I think